src/libraries/System.Text.Json/src/System/Text/Json/JsonHelpers.Escaping.cs

```diff
@@ -53,6 +53,40 @@ public static byte[] GetEscapedPropertyNameSection(ReadOnlySpan<byte> utf8Value,
             return escapedString;
         }
 
+        public static string GetPropertyName(
+            string rawPropertyName,
+            JavaScriptEncoder? encoder)
+        {
+            int indexOfFirstCharacterToEncode;
+            string returnValue;
+
+            indexOfFirstCharacterToEncode = JsonWriterHelper.NeedsEscaping(rawPropertyName.AsSpan(), encoder);
+
+            if (indexOfFirstCharacterToEncode != -1)
+            {
+                returnValue = GetEscapedPropertyName(rawPropertyName);
+            }
+            else
+            {
+                returnValue = rawPropertyName;
+            }
+
+            return returnValue;
+        }
+
+        private static string GetEscapedPropertyName(string rawPropertyName)
+        {
+            const string SingleQuote = "'";
+            const string EscapedSingleQuote = "\\'";
+
+            string returnValue;
+
+            returnValue = rawPropertyName
+                .Replace(SingleQuote, EscapedSingleQuote);
```

Comment:
I wanted to use here [Regex.Replace(String, String, String, RegexOptions)](https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.replace?view=net-7.0#system-text-regularexpressions-regex-replace(system-string-system-string-system-string-system-text-regularexpressions-regexoptions)), but I could not understand how to use System.Text.RegularExpressions assembly from current.